### PR TITLE
Series with differentiation

### PR DESCRIPTION
**Pull request by isuruf**
I tried to get the same output as SymPy for gamma with this and had to change some of the functions.

TODO
Add tests for,
- [x] polygamma diff
- [x] polygamma subs
- [x] gamma subs
- [x] zeta auto simplification
- [x] polygamma auto simplification
- [x] harmonic

### CODE REVIEW

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -403,6 +402,12 @@ static RCP<const Basic> diff(const CLASS &self, \
         return mul(mul(self.rcp_from_this(), polygamma(zero, gamma_arg)), gamma_arg->diff(x));
     }
 
+    static RCP<const Basic> diff(const PolyGamma &self,
+            const RCP<const Symbol> &x) {
+        auto args = self.get_args();
+        return polygamma(add(args[0], one), args[1]);
```
Comment (nishnik) [inline symengine/derivative.cpp]: I think it should be like [this](https://github.com/nishnik/symengine/blob/ba6ccd834301622555a6e2936c6a42994cfaa02c/symengine/derivative.cpp#L435).

Comment (isuruf) [inline symengine/derivative.cpp]: No, both are incorrect. I'll push the changes

**Comment by isuruf** [comment]
Appveyor failure is due to a network failure. I've restarted the appveyor build

**Comment by certik** [comment]
It looks good. It needs a merge with master first.

**Comment by isuruf** [comment]
Merged with master.

**Comment by isuruf** [comment]
Merged again

**Comment by certik** [comment]
@ping me once you want me to review.

**Comment by isuruf** [comment]
It's ready for review

**Comment by certik** [comment]
Ok, that looks good to me. +1 to merge if tests pass.

**Comment by isuruf** [comment]
Fixed failures

**Comment by rwst** [comment]
I reviewed the last three commits, so the review should be complete.

**Pull request merged**